extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/PrometheusMeterRegistryProducer.java

```diff
@@ -1,25 +1,33 @@
 package io.quarkus.micrometer.runtime.export;
 
+import javax.enterprise.inject.Alternative;
+import javax.enterprise.inject.Instance;
 import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
 import javax.inject.Singleton;
 import javax.interceptor.Interceptor;
 
 import io.micrometer.core.instrument.Clock;
 import io.micrometer.prometheus.PrometheusConfig;
 import io.micrometer.prometheus.PrometheusMeterRegistry;
 import io.prometheus.client.CollectorRegistry;
-import io.quarkus.arc.AlternativePriority;
+import io.prometheus.client.exemplars.ExemplarSampler;
+import io.quarkus.arc.Priority;
 
 /**
  * This producer is only registered if the {@code quarkus.micrometer.export.prometheus.default-registry} is set to {@code true}.
  */
 public class PrometheusMeterRegistryProducer {
 
+    @Inject
+    Instance<ExemplarSampler> sampler;
+
     @Produces
     @Singleton
-    @AlternativePriority(Interceptor.Priority.APPLICATION + 100)
+    @Alternative
+    @Priority(Interceptor.Priority.APPLICATION + 100)
```

Comment:
I very much doubt that this is the place to add this tracing data related with the min, max...
I would expect this to be added as tags in the particular metric and not in the exporters.

Reply:
I just followed https://github.com/quarkusio/quarkus/issues/28208#issuecomment-1262135629. If we want to do it an other way, I will have to re-think how we can achieve the same.

Reply:
Even as it is right now, the VertxHttpServerMetrics don't get there span and trace ids, as `VertxMDC.INSTANCE` has already be cleaned when exemplar sampling happens.

Reply:
We would need to collect the trace ID of each sample and preserve the ones regarding the min and the max.

Reply:
Tags for min/max are not what exemplars are, and there is a clear separation of responsibility here: when we call record, we need a way to help record the span (which is what we're doing here with the span provider). It's up to the prometheus registry client to do the rest.. 

https://github.com/micrometer-metrics/micrometer/pull/3091#issue-1177411405

Reply:
@ebullient This code works for annotated methods, but does not for VertxHttpServerMetrics, because the MDC is already cleaned, when the `sample` are stopped, do you think there is anything we can do about it ?

Reply:
Tests are in dedicated integration-test module